File: DSA/Vikash_kumar/Day_12_Question_2.py
#Day12-Question2-
def sum_digits(number):
    if number < 10:
        return number
    else:
        # Sum the digits of the number
        digit_sum = sum(int(digit) for digit in str(number))
        # Recursively call the function with the new sum
        return sum_digits(digit_sum)

n = int(input("Enter a number: ")
print(f"The single digit result for {n} is {sum_digits(n)}")

File: DSA/Manav-Rajput/Day12_question_2.java
public class DigitSum {
    public static int sumDigits(int number) {
        if (number < 10) {
            return number;
        } else {
            return sumDigits(sumOfDigits(number));
        }
    }
    public static int sumOfDigits(int number) {
        int sum = 0;
        while (number != 0) {
            sum += number % 10;  
            number /= 10;        
        }
        return sum;
    }
    public static void main(String[] args) {
        int number = 9875;  
        int result = sumDigits(number);
        System.out.println("The single digit result of the number " + number + " is: " + result);
    }
}
